bpf/bpf_lxc.c

```diff
@@ -595,10 +598,13 @@ static inline int __inline__ ipv4_policy(struct __sk_buff *skb, int ifindex, __u
 	if (ret < 0)
 		return ret;
 
-	if (policy_can_access(&POLICY_MAP, skb, src_label) != TC_ACT_OK) {
-		if (ret != CT_ESTABLISHED && ret != CT_REPLY && ret != CT_RELATED)
+	/* Policy lookup is done on every packet to account for packets that
+	 * passed through the allowed consumer. */
+	verdict = policy_can_access(&POLICY_MAP, skb, src_label);
+	if (unlikely(ret == CT_NEW)) {
+		if (verdict != TC_ACT_OK)
```

Comment:
It leads to a better fast path because `CT_NEW` should only be seen for the first packet of a flow. We only need to look at the policy verdict for new flows. All packets have to go through the policy table though for accounting. We can still optimize this by making the account part optional and it that case only perform the map lookup when the flow is first created.